spec/requests/api/move_events_controller_lockout_spec.rb

```diff
@@ -108,18 +79,68 @@
 
     context 'with validation errors' do
       context 'with a bad timestamp' do
-        let(:lockout_params) { { data: { type: 'lockouts', attributes: { timestamp: 'Foo-Bar' } } } }
+        let(:lockout_params) do
+          { data: { type: 'lockouts',
+                    attributes: { timestamp: 'Foo-Bar' },
+                    relationships: {
+                      from_location: { data: { type: 'locations', id: lockout_location.id } },
+                    } } }
+        end
 
         it_behaves_like 'an endpoint that responds with error 422' do
-          let(:errors_422) { [{ 'title' => 'Invalid timestamp', 'detail' => 'Validation failed: Timestamp must be formatted as a valid ISO-8601 date-time' }] }
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid timestamp',
+              'detail' => 'Validation failed: Timestamp must be formatted as a valid ISO-8601 date-time',
+            }]
+          end
         end
       end
 
       context 'with a bad event type' do
         let(:lockout_params) { { data: { type: 'Foo-bar', attributes: { timestamp: '2020-04-23T18:25:43.511Z' } } } }
 
         it_behaves_like 'an endpoint that responds with error 422' do
-          let(:errors_422) { [{ 'title' => 'Invalid type', 'detail' => 'Validation failed: Type is not included in the list' }] }
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid type',
+              'detail' => 'Validation failed: Type is not included in the list',
+            }]
+          end
+        end
+      end
+
+      context 'with a missing from_location' do
+        let(:lockout_params) { { data: { type: 'lockouts', attributes: { timestamp: '2020-04-23T18:25:43.511Z' } } } }
+
+        it_behaves_like 'an endpoint that responds with error 422' do
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid from_location',
+              'detail' => 'Validation failed: From location is missing',
+            }]
+          end
+        end
+      end
+
+      context 'with a non-existent from_location' do
+        let(:lockout_params) do
+          {
+            data: {
+              type: 'lockouts',
+              attributes: { timestamp: '2020-04-23T18:25:43.511Z' },
+              relationships: { from_location: { data: { type: 'locations', id: 'atlantis' } } },
+            },
+          }
+        end
+
+        it_behaves_like 'an endpoint that responds with error 422' do
+          let(:errors_422) do
+            [{
+              'title' => 'Invalid from_location',
+              'detail' => 'Validation failed: From location was not found',
+            }]
```

Comment:
we are testing 2 times the 422 in this spec (which is a bit expensive to do here),
maybe we can test this case elsewhere?

Reply:
ok, I have moved the missing location to the params validator spec